src/components/tx-flow/common/TxNonce/index.tsx

```diff
@@ -51,34 +51,56 @@ const NonceFormOption = memo(function NonceFormOption({
 })
 
 const TxNonce = () => {
-  const [error, setError] = useState<boolean>(false)
+  const [error, setError] = useState<string>()
   const { safe } = useSafeInfo()
   const previousNonces = usePreviousNonces()
   const { nonce, setNonce, safeTx, recommendedNonce } = useContext(SafeTxContext)
   const isEmpty = useRef<boolean>(false)
   const isEditable = !safeTx || safeTx?.signatures.size === 0
   const readonly = !isEditable
 
-  const isValidInput = useCallback(
+  const validateInput = useCallback(
     (value: string | AutocompleteValue<unknown, false, false, false>) => {
-      return Number(value) >= safe.nonce
+      const nonce = Number(value)
+
+      if (isNaN(nonce)) {
+        return 'Nonce must be a number'
+      }
+
+      if (nonce < safe.nonce) {
+        return `Nonce can't be lower than ${safe.nonce}`
+      }
+
+      if (nonce >= Number.MAX_SAFE_INTEGER) {
+        return 'Nonce is too big'
+      }
     },
     [safe.nonce],
   )
 
   const handleChange = useCallback(
     (_e: SyntheticEvent, value: string | AutocompleteValue<unknown, false, false, false>) => {
+      const isNotValid = validateInput(value)
+
+      if (isNotValid) {
+        setError(validateInput(value))
+        isEmpty.current = false
+        return
+      }
+
+      setError(undefined)
       isEmpty.current = value === ''
-      const nonce = Number(value)
-      if (isNaN(nonce)) return
-      setError(!isValidInput(value))
-      setNonce(nonce)
+      setNonce(Number(value))
     },
-    [isValidInput, setNonce],
+    [validateInput, setNonce],
   )
 
+  const handleBlur = useCallback(() => {
+    setError(undefined)
```

Comment:
What if the current `nonce` is erroneous?

Reply:
Since we don't save the nonce if its incorrect the current nonce must be correct so we can safely remove the error.